ee_plugin/processing/add_image_collection.py

```diff
@@ -466,24 +476,41 @@ def processAlgorithm(self, parameters, context, feedback):
                     )
 
         # Apply compositing logic
-        compositing_dict = {
-            "Mean": ic.mean(),
-            "Max": ic.max(),
-            "Min": ic.min(),
-            "Median": ic.median(),
-            "Percentile": ic.reduce(ee.Reducer.percentile([percentile_value])),
-        }
-
-        ic = compositing_dict.get(compositing_method, ic.mosaic())
```

Comment:
@spatialthoughts; I think I caught the performance issue here. We were pre-computing every single option. Definitely an oversight. Things seem a lot more responsive on my side now, would love for you to check it out.